src/System.Security.Claims/src/System/Security/Claims/ClaimsIdentity.cs

```diff
@@ -0,0 +1,970 @@
+            {
+                throw new ArgumentNullException("other");
+            }
+
+            if (other._actor != null)
+            {
+                _actor = other._actor.Clone();
+            }
+
+            _authenticationType = other._authenticationType;
+            _bootstrapContext = other._bootstrapContext;
+            _label = other._label;
+            _nameClaimType = other._nameClaimType;
+            _roleClaimType = other._roleClaimType;
+            if (other._userSerializationData != null)
+            {
+                _userSerializationData = other._userSerializationData.Clone() as byte[];
+            }
+
+            SafeAddClaims(other._instanceClaims);
+        }
+
+        /// <summary>
+        /// Gets the authentication type that can be used to determine how this <see cref="ClaimsIdentity"/> authenticated to an authority.
+        /// </summary>
+        public virtual string AuthenticationType
+        {
+            get { return _authenticationType; }
+        }
+
+        /// <summary>
+        /// Gets a value that indicates if the user has been authenticated.
+        /// </summary>
+        public virtual bool IsAuthenticated
+        {
+            get { return !string.IsNullOrEmpty(_authenticationType); }
+        }
+
+        /// <summary>
+        /// Gets or sets a <see cref="ClaimsIdentity"/> that was granted delegation rights.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">if 'value' results in a circular refrence back to 'this'.</exception>
+        public ClaimsIdentity Actor
+        {
+            get { return _actor; }
+            set
+            {
+                if (value != null)
+                {
+                    if (IsCircular(value))
+                    {
+                        throw new InvalidOperationException(SR.GetResourceString("InvalidOperationException_ActorGraphCircular", "Actor cannot be set so that circular directed graph will exist chaining the subjects together."));
+                    }
+                }
+                _actor = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a context that was used to create this <see cref="ClaimsIdentity"/>.
+        /// </summary>
+        public object BootstrapContext
+        {
+            get { return _bootstrapContext; }
+
+            [SecurityCritical]
+            set
+            { _bootstrapContext = value; }
+        }
+
+        /// <summary>
+        /// Gets the claims as <see cref="IEnumerable{Claim}"/>, associated with this <see cref="ClaimsIdentity"/>.
+        /// </summary>       
+        /// <remarks>May contain nulls.
+        public virtual IEnumerable<Claim> Claims
+        {
+            get
+            {
+                for (int i = 0; i < _instanceClaims.Count; i++)
+                {
+                    yield return _instanceClaims[i];
+                }
+
+                if (_externalClaims != null)
```

Comment:
It's used so rarely and was only put in to ensure that "IsInRole(x) iff a matching 'role' claim can be found". This is tied to RolePrincipal.
We should think about it this still matters and perhaps remove the _externalClaims logic.